common/kernel.c

```diff
@@ -27,16 +27,34 @@
 #include <lib.h>
 #include <multiboot.h>
 #include <percpu.h>
+#include <processor.h>
 #include <sched.h>
 #include <setup.h>
 
+/* relative difference (in size of a memory address) on the stack
+    between flags and return address as pushed by usermode_call_asm */
+#define RET2KERN_FLAGS_SP_DIFF 3
```

Comment:
It would be better to additionally push kernel flags before IRET frame and just pop it when the kernel stack is restored.

Reply:
Good idea, thanks!